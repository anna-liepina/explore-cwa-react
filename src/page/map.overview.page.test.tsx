import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import MapOverviewPage, { IMapOverviewPageProps } from './map.overview.page';

describe('<MapOverviewPage/>', () => {
    const MapPage = (props: IMapOverviewPageProps ) => {
        return (
            <MemoryRouter>
                <MapOverviewPage {...props} />
            </MemoryRouter>
        )
    }

    const props = {
        form: { config: [] }
    };

    const optionalProps = {
        'data-cy': 'optProps.data-cy',
    };

    describe('render', () => {
        it('with default/required props', () => {
            const { asFragment } = render(<MapPage {...props} />);

            expect(asFragment()).toMatchSnapshot();
        });

        it('with optional/required props', () => {
            const { asFragment } = render(<MapPage {...props} {...optionalProps} />);

            expect(asFragment()).toMatchSnapshot();
        });
    });
});
