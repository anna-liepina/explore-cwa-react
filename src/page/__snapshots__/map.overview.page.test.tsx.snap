// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapOverviewPage/> render with default/required props 1`] = `
<DocumentFragment>
  <section
    class="map-handler"
    style="height: 678px;"
  >
    <form
      class="form"
    >
      <div
        class="form_controls"
      />
    </form>
    <div
      class="map-handler--legend"
    >
      <h4>
        map legend
      </h4>
      <ul>
        <li>
          <div
            class="pigeon-click-block"
            style="position: absolute; transform: translate(NaNpx, NaNpx); pointer-events: none; cursor: pointer;"
          >
            <svg
              fill="none"
              height="34"
              viewBox="0 0 61 71"
              width="29"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                style="pointer-events: auto;"
              >
                <path
                  d="M52 31.5C52 36.8395 49.18 42.314 45.0107 47.6094C40.8672 52.872 35.619 57.678 31.1763 61.6922C30.7916 62.0398 30.2084 62.0398 29.8237 61.6922C25.381 57.678 20.1328 52.872 15.9893 47.6094C11.82 42.314 9 36.8395 9 31.5C9 18.5709 18.6801 9 30.5 9C42.3199 9 52 18.5709 52 31.5Z"
                  fill="#f00"
                  stroke="white"
                  stroke-width="4"
                />
                <circle
                  cx="30.5"
                  cy="30.5"
                  fill="white"
                  opacity="0.6"
                  r="8.5"
                />
              </g>
            </svg>
          </div>
           incidents reported by police
        </li>
        <li>
          <div
            class="pigeon-click-block"
            style="position: absolute; transform: translate(NaNpx, NaNpx); pointer-events: none; cursor: pointer;"
          >
            <svg
              fill="none"
              height="34"
              viewBox="0 0 61 71"
              width="29"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                style="pointer-events: auto;"
              >
                <path
                  d="M52 31.5C52 36.8395 49.18 42.314 45.0107 47.6094C40.8672 52.872 35.619 57.678 31.1763 61.6922C30.7916 62.0398 30.2084 62.0398 29.8237 61.6922C25.381 57.678 20.1328 52.872 15.9893 47.6094C11.82 42.314 9 36.8395 9 31.5C9 18.5709 18.6801 9 30.5 9C42.3199 9 52 18.5709 52 31.5Z"
                  fill="#000"
                  stroke="white"
                  stroke-width="4"
                />
                <circle
                  cx="30.5"
                  cy="30.5"
                  fill="white"
                  opacity="0.6"
                  r="8.5"
                />
              </g>
            </svg>
          </div>
           property sales
        </li>
      </ul>
    </div>
  </section>
</DocumentFragment>
`;

exports[`<MapOverviewPage/> render with optional/required props 1`] = `
<DocumentFragment>
  <section
    class="map-handler"
    style="height: 678px;"
  >
    <form
      class="form"
    >
      <div
        class="form_controls"
      />
    </form>
    <div
      class="map-handler--legend"
    >
      <h4>
        map legend
      </h4>
      <ul>
        <li>
          <div
            class="pigeon-click-block"
            style="position: absolute; transform: translate(NaNpx, NaNpx); pointer-events: none; cursor: pointer;"
          >
            <svg
              fill="none"
              height="34"
              viewBox="0 0 61 71"
              width="29"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                style="pointer-events: auto;"
              >
                <path
                  d="M52 31.5C52 36.8395 49.18 42.314 45.0107 47.6094C40.8672 52.872 35.619 57.678 31.1763 61.6922C30.7916 62.0398 30.2084 62.0398 29.8237 61.6922C25.381 57.678 20.1328 52.872 15.9893 47.6094C11.82 42.314 9 36.8395 9 31.5C9 18.5709 18.6801 9 30.5 9C42.3199 9 52 18.5709 52 31.5Z"
                  fill="#f00"
                  stroke="white"
                  stroke-width="4"
                />
                <circle
                  cx="30.5"
                  cy="30.5"
                  fill="white"
                  opacity="0.6"
                  r="8.5"
                />
              </g>
            </svg>
          </div>
           incidents reported by police
        </li>
        <li>
          <div
            class="pigeon-click-block"
            style="position: absolute; transform: translate(NaNpx, NaNpx); pointer-events: none; cursor: pointer;"
          >
            <svg
              fill="none"
              height="34"
              viewBox="0 0 61 71"
              width="29"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                style="pointer-events: auto;"
              >
                <path
                  d="M52 31.5C52 36.8395 49.18 42.314 45.0107 47.6094C40.8672 52.872 35.619 57.678 31.1763 61.6922C30.7916 62.0398 30.2084 62.0398 29.8237 61.6922C25.381 57.678 20.1328 52.872 15.9893 47.6094C11.82 42.314 9 36.8395 9 31.5C9 18.5709 18.6801 9 30.5 9C42.3199 9 52 18.5709 52 31.5Z"
                  fill="#000"
                  stroke="white"
                  stroke-width="4"
                />
                <circle
                  cx="30.5"
                  cy="30.5"
                  fill="white"
                  opacity="0.6"
                  r="8.5"
                />
              </g>
            </svg>
          </div>
           property sales
        </li>
      </ul>
    </div>
  </section>
</DocumentFragment>
`;
